raster/r.texture/testsuite/test_texture.py

```diff
@@ -24,59 +25,164 @@ def setUpClass(cls):
     def tearDownClass(cls):
         cls.del_temp_region()
         cls.runModule("g.remove", flags="f", type="raster", name="asm_ASM")
+        cls.runModule("g.remove", flags="f", type="raster", name="contrast_Contr")
         cls.runModule("g.remove", flags="f", type="raster", name="corr_Corr")
-        cls.runModule("g.remove", flags="f", type="raster", name="sa_SA")
         cls.runModule("g.remove", flags="f", type="raster", name="var_Var")
         cls.runModule("g.remove", flags="f", type="raster", name="idm_IDM")
+        cls.runModule("g.remove", flags="f", type="raster", name="var_Var")
+        cls.runModule("g.remove", flags="f", type="raster", name="sa_SA")
+        cls.runModule("g.remove", flags="f", type="raster", name="sv_SV")
+        cls.runModule("g.remove", flags="f", type="raster", name="se_SE")
+        cls.runModule("g.remove", flags="f", type="raster", name="entr_Entr")
+        cls.runModule("g.remove", flags="f", type="raster", name="dv_DV")
+        cls.runModule("g.remove", flags="f", type="raster", name="de_DE")
+        cls.runModule("g.remove", flags="f", type="raster", name="moc1_MOC-1")
+        cls.runModule("g.remove", flags="f", type="raster", name="moc2_MOC-2")
 
     def test_asm(self):
         """Testing method asm"""
         asm_ASM = "asm_ASM"
-        string = """min=1
-        max=1
+        values = """min=-0.0455724261701107
+        max=0.0380486063659191
+        mean=-0.000136686790876467
+        variance=0.000158102
         cells=1000246"""
         self.assertModule("r.texture", input=self.input, output="asm", method="asm")
-        self.assertRasterFitsUnivar(asm_ASM, reference=string, precision=2)
+        self.assertRasterFitsUnivar(asm_ASM, reference=values, precision=2)
+
+    def test_contrast(self):
+        """Testing method contrast"""
+        corr_output = "contrast_Contr"
+        values = """min=0
+        max=14680.8
+        mean=176.569
+        variance=185767
+        cells=1000246"""
```

Comment:
Better than cells is probably n (which is count). There are strong guarantees in GRASS GIS (or at least very high chance) that a tool will generate the right size of a raster, so there is not a great need to test cell count (but you can since you have it there already). However, n tests the non-null cells which tells us if the tool is creating some unexpected nulls in the computation.